diff --git a/src/src/tbb/CMakeLists.txt b/src/src/tbb/CMakeLists.txt
index c67b8bb..788a1ff 100644
--- a/src/src/tbb/CMakeLists.txt
+++ b/src/src/tbb/CMakeLists.txt
@@ -137,7 +137,7 @@ else()
     if (BUILD_TYPE STREQUAL "debug")
         set(TBB_LIB_NAME ${TBB_LIB_NAME}${CMAKE_DEBUG_POSTFIX})
     endif()
-    configure_file(${CMAKE_SOURCE_DIR}/integration/pkg-config/tbb.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${TBB_LIB_NAME}.pc @ONLY)
+    configure_file(${CMAKE_SOURCE_DIR}/src/integration/pkg-config/tbb.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${TBB_LIB_NAME}.pc @ONLY)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${TBB_LIB_NAME}.pc
             DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
