--- a/src/src/test/harness.h
+++ b/src/src/test/harness.h
@@ -540,7 +540,7 @@ public:
         // calls to pthread_create fail with out-of-memory error.
         // Therefore we set the stack size explicitly (as for TBB worker threads).
 #if !defined(HARNESS_THREAD_STACK_SIZE)
-#if __i386__||__i386||__arm__
+#if __i386__||__i386||__arm__||(_MIPS_ISA == _MIPS_ISA_MIPS32)
         const size_t stack_size = 1*MByte;
 #elif __x86_64__
         const size_t stack_size = 2*MByte;
