diff --git a/src/src/test/test_parallel_for.cpp b/src/src/test/test_parallel_for.cpp
index fb17c6e..6fe76e9 100644
--- a/src/src/test/test_parallel_for.cpp
+++ b/src/src/test/test_parallel_for.cpp
@@ -662,7 +662,7 @@ range_borders combine_range(const MapType& map) {
 template <typename Partitioner>
 void test_body() {
     unsigned hw_concurrency = tbb::tbb_thread::hardware_concurrency();
-    for (unsigned int num_threads = hw_concurrency / 4 + 1; num_threads < hw_concurrency; num_threads *= 2) {
+    for (unsigned int num_threads = hw_concurrency / 2 + 1; num_threads < hw_concurrency; num_threads *= 2) {
         REMARK("  num_threads=%lu\n", num_threads);
         for (size_t range_begin = 0, range_end = num_threads * 10 - 1, i = 0; i < 3;
              range_begin += num_threads, range_end += num_threads + 1, ++i) {
