diff --git a/src/src/test/test_task_scheduler_init.cpp b/src/src/test/test_task_scheduler_init.cpp
index 4b596dd..2c1e8c8 100644
--- a/src/src/test/test_task_scheduler_init.cpp
+++ b/src/src/test/test_task_scheduler_init.cpp
@@ -285,7 +285,7 @@ namespace TestBlockingTerminateNS {
             }
         };
         void operator()() {
-            const int numThreads = 4;
+            const int numThreads = tbb::tbb_thread::hardware_concurrency();
             tbb::task_scheduler_init init( numThreads );
             Harness::SpinBarrier barrier( numThreads );
             tbb::parallel_for( 0, numThreads, Body( barrier ) );
